src/views/Pools/components/Vault/VaultStakeButtonGroup.tsx

```diff
@@ -18,14 +20,18 @@ export const VaultStakeButtonGroup = ({
     </Box>,
     {},
   )
-
   return (
     <Box width="100%">
       <FlexGap gap="12px">
         <Button style={{ flex: 1 }} onClick={onFlexibleClick}>
           {t('Flexible')}
         </Button>
-        <Button style={{ flex: 1 }} disabled decorator={{ text: t('Soon') }} onClick={onLockedClick}>
+        <Button
+          style={{ flex: 1 }}
+          disabled={maxDuration?.toNumber() === 0}
+          decorator={maxDuration?.toNumber() === 0 && { text: t('Soon') }}
```

Comment:
Disable when maxDuration is undefined, and only show decorator when maxDuration has value and is zero.
You can just use
maxDuration?.isZero()

Reply:
fixed